pkg/postage/postagecontract/contract.go

```diff
@@ -204,6 +242,42 @@ func (c *postageContract) CreateBatch(ctx context.Context, initialBalance *big.I
 	return nil, ErrBatchCreate
 }
 
+func (c *postageContract) TopUpBatch(ctx context.Context, batchID []byte, topUpAmount *big.Int) error {
+
+	batch, err := c.postageStorer.Get(batchID)
+	if err != nil {
+		return err
+	}
+
+	totalAmount := big.NewInt(0).Mul(topUpAmount, big.NewInt(int64(1<<batch.Depth)))
+	balance, err := c.getBalance(ctx)
+	if err != nil {
+		return err
+	}
+
+	if balance.Cmp(totalAmount) < 0 {
+		return ErrInsufficientFunds
+	}
+
+	_, err = c.sendApproveTransaction(ctx, totalAmount)
+	if err != nil {
+		return err
+	}
+
+	receipt, err := c.sendTopUpBatchTransaction(ctx, batch.ID, totalAmount)
+	if err != nil {
+		return err
+	}
+
+	for _, ev := range receipt.Logs {
+		if ev.Address == c.postageContractAddress && ev.Topics[0] == batchTopUpTopic {
```

Comment:
this will panic if there is an event. with no topics. should never happen with the real contract but that should be handled nonetheless. Ideally use. `transaction.FindSingleEvent` which already has this.

Reply:
So the `FindSingleEvent` parses the event as well. I did not want to do that as then we would have to define a struct locally for the event and we dont really need to check the contents of the event here.

So I have added a check for `len(ev.Topics) > 0`